### ISSUE TITLE
connection to the device ends after 20-30 minutes.

### ISSUE THREAD

**Issue by KGQAJH** [doronz88/pymobiledevice3#1015]
When using the "python3 -m pymobiledevice3 developer dvt sysmon process single" command, the connection to the device ends after 20-30 minutes.

**Comment by doronz88** [comment]
Please provide more context and steps to reproduce

**Comment by KGQAJH** [comment]
hello

Literally run the app on an iOS device with iOS 16 version, use the pymobiledevice3 module, and measure CPU, memory, fps, and temperature with the commands below.

""python3 -m pymobiledevice3 developer dvt sysmon process single"

After 20-30 minutes, fps and temperature are continuously output, but cpu and memory are not output.

pymobiledevice3 version = 4.72

**Comment by doronz88** [comment]
This command is a oneshot. Are you running it over and over from a shell script?

**Comment by KGQAJH** [comment]
Yes, I use it repeatedly. We want to collect the device's cpu and mem values ​​every second in real time.   

 def get_ios_cpu_mem(self, name):
        command = f"ython3 -m pymobiledevice3 developer dvt sysmon process single --attributes name={name} --no-color"
        process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
                                   encoding='utf-8')

        stdout, stderr = process.communicate()

        if process.returncode != 0:
            logging.error(f"Command failed with error: {stderr.strip()}")
            logging.error(f"Full stdout: {stdout.strip()}")
            return 0, 0  # 기본값으로 0 반환

        if not stdout.strip():
            logging.error("Command returned empty output.")
            return 0, 0  # 기본값으로 0 반환

        try:
            data = json.loads(stdout)
            if not data:  # 데이터가 비어 있을 경우 처리
                logging.error("Output data is empty")
                return 0, 0  # 기본값으로 0 반환
        except json.JSONDecodeError as e:
            logging.error(f"Failed to parse JSON from command output: {e}")
            logging.error(f"Full stdout: {stdout.strip()}")
            return 0, 0  # 기본값으로 0 반환

        # 값 추출
        cpuUsage = data[0].get("cpuUsage", 0)  # 기본값 0 설정
        physFootprint = data[0].get("physFootprint", 0)  # 기본값 0 설정
        time.sleep(1)

        return cpuUsage, physFootprint

**Comment by sunghyun123** [comment]
I also had the same problem as above. Please tell me how to solve it

**Comment by doronz88** [comment]
Answered: `sysmon ... single` is a one-shot command; it was being looped, which caused the apparent disconnects. Closing.
